archunit/src/main/java/com/tngtech/archunit/library/modules/ArchModules.java

```diff
@@ -0,0 +1,556 @@
+            }
+
+            @Override
+            public Identifier associate(JavaClass javaClass) {
+                return packageToIdentifier.entrySet().stream()
+                        .filter(it -> packageContains(it.getKey(), javaClass.getPackageName()))
+                        .findFirst()
+                        .map(Map.Entry::getValue)
+                        .orElse(Identifier.ignore());
+            }
+
+            JavaClass getRootClassOf(Identifier identifier) {
+                return identifierToRootClass.get(identifier);
+            }
+        }
+    }
+
+    /**
+     * An element of the fluent API to create {@link ArchModules}
+     */
+    @PublicAPI(usage = ACCESS)
+    public static class Creator {
+        private final IdentifierAssociation identifierAssociation;
+        private final Function<Identifier, String> deriveNameFunction;
+
+        private Creator(IdentifierAssociation identifierAssociation) {
+            this(identifierAssociation, DEFAULT_NAMING_STRATEGY);
+        }
+
+        private Creator(IdentifierAssociation identifierAssociation, Function<Identifier, String> deriveNameFunction) {
+            this.identifierAssociation = checkNotNull(identifierAssociation);
+            this.deriveNameFunction = checkNotNull(deriveNameFunction);
+        }
+
+        /**
+         * Allows to customize each {@link ArchModule} {@link ArchModule#getName() name} by specifying a string pattern
+         * that defines how to derive the name from the {@link ArchModule.Identifier}.<br>
+         * In particular, the passed {@code namingPattern} may contain numbered placeholders like <code>${1}</code>
+         * to refer to parts from the {@link ArchModule.Identifier}. It may also contain the special placeholder
+         * <code>$@</code> to refer to the colon-joined form of the identifier.<br>
+         * Suppose the {@link ArchModule.Identifier} is {@code ["customer", "creation"]}, then the name could be derived as
+         * <pre><code>
+         * "Module[${1}/${2}]" -> would yield the name "Module[customer/creation]"
+         * "Module[$@]         -> would yield the name "Module[customer:creation]"
+         * </code></pre>
+         * Note that the derived name must be unique between all {@link ArchModule modules}.
+         *
+         * @param namingPattern A string naming pattern deriving the {@link ArchModule} {@link ArchModule#getName() name} from
+         *                      the {@link ArchModule.Identifier}
+         * @return A fluent API to further customize how to create {@link ArchModules}
+         */
+        @PublicAPI(usage = ACCESS)
+        public Creator deriveNameFromPattern(final String namingPattern) {
+            return new Creator(identifierAssociation, identifier -> {
+                String result = namingPattern;
+                for (int i = 1; i <= identifier.getNumberOfParts(); i++) {
```

Comment:
Yes, true in theory, but in this case I would wait for somebody to really complain about this before calling to action :joy: Because in this very specific context I can hardly imagine it... And if somebody would really encounter this they would have the more generic API to fall back to to solve their issue.